configuration/set.go

```diff
@@ -0,0 +1,171 @@
+		ConfigLocation = filepath.Join(HomeDir, types.OssIndexDirName, types.OssIndexConfigFileName)
+		err = getAndSetOSSIndexConfig(reader)
+	case "":
+		return
+	default:
+		LogLady.Info("User chose to set OSS Index config, moving forward")
+		fmt.Println("Invalid value, 'iq' and 'ossindex' are accepted values, try again!")
+		GetConfigFromCommandLine(stdin)
+	}
+
+	if err != nil {
+		LogLady.Error(err)
+		return
+	}
+	return
+}
+
+func getAndSetIQConfig(reader *bufio.Reader) (err error) {
+	LogLady.Info("Getting config for IQ Server from user")
+
+	iqConfig := IQConfig{Server: "http://localhost:8070", Username: "admin", Token: "admin123"}
+
+	fmt.Print("What is the address of your Nexus IQ Server (default: http://localhost:8070)? ")
+	server, _ := reader.ReadString('\n')
+	iqConfig.Server = emptyOrDefault(server, iqConfig.Server)
+
+	fmt.Print("What username do you want to authenticate as (default: admin)? ")
+	username, _ := reader.ReadString('\n')
+	iqConfig.Username = emptyOrDefault(username, iqConfig.Username)
+
+	fmt.Print("What token do you want to use (default: admin123)? ")
+	token, _ := reader.ReadString('\n')
+	iqConfig.Token = emptyOrDefault(token, iqConfig.Token)
+
+	if iqConfig.Username == "admin" || iqConfig.Token == "admin123" {
+		LogLady.Info("Warning user of bad life choices, using default values for IQ Server username or token")
+		warnUserOfBadLifeChoices()
+		fmt.Print("[y/N]? ")
+		theChoice, _ := reader.ReadString('\n')
+		theChoice = emptyOrDefault(theChoice, "y")
+		if theChoice == "y" {
+			LogLady.Info("User chose to rectify their bad life choices, asking for config again")
+			getAndSetIQConfig(reader)
+		}
+	}
+
+	LogLady.Info("Successfully got IQ Server config from user, attempting to save to disk")
+	err = marshallAndWriteToDisk(iqConfig)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func emptyOrDefault(value string, defaultValue string) string {
+	str := strings.Trim(strings.TrimSpace(value), "\n")
+	if str == "" {
+		return defaultValue
+	}
+	return str
+}
+
+func getAndSetOSSIndexConfig(reader *bufio.Reader) (err error) {
+	LogLady.Info("Getting config for OSS Index from user")
+
+	ossIndexConfig := OSSIndexConfig{}
+
+	fmt.Print("What username do you want to authenticate as (ex: admin)? ")
+	ossIndexConfig.Username, _ = reader.ReadString('\n')
+	ossIndexConfig.Username = strings.Trim(strings.TrimSpace(ossIndexConfig.Username), "\n")
+
+	fmt.Print("What token do you want to use? ")
+	ossIndexConfig.Token, _ = reader.ReadString('\n')
+	ossIndexConfig.Token = strings.Trim(strings.TrimSpace(ossIndexConfig.Token), "\n")
+
+	LogLady.Info("Successfully got OSS Index config from user, attempting to save to disk")
+	err = marshallAndWriteToDisk(ossIndexConfig)
+	if err != nil {
+		LogLady.Error(err)
+		return
+	}
+
+	return
+}
+
+func marshallAndWriteToDisk(config interface{}) (err error) {
+	d, err := yaml.Marshal(config)
+	if err != nil {
+		return err
+	}
+
+	base := filepath.Dir(ConfigLocation)
```

Comment:
Added @todo for Issue #111 : use `io.Writer` instead of `ConfigLocation` inside `configuration/set.go -> marshallAndWriteToDisk`.

I have Readers/Writers on the brain. ;)